src/Atmos/Model/AtmosModel.jl

```diff
@@ -552,6 +557,13 @@ include("linear_atmos_tendencies.jl")
 include("atmos_tendencies.jl")        # specify atmos tendencies
 include("get_prognostic_vars.jl")     # get tuple of prognostic variables
 
+
+sub_model(atmos::AtmosModel, ::Type{<:AbstractEnergyModel}) =
+    energy_model(atmos)
+sub_model(atmos::AtmosModel, ::Type{<:AbstractMoistureModel}) =
+    moisture_model(atmos)
+
+
```

Comment:
Shouldn't these be returning tuples?

```suggestion
sub_model(atmos::AtmosModel, ::Type{<:AbstractEnergyModel}) =
    (energy_model(atmos),)
sub_model(atmos::AtmosModel, ::Type{<:AbstractMoistureModel}) =
    (moisture_model(atmos),)

```
Also, although not currently used, we should define `sub_model` for all the `AtmosModel` sub-models.

Reply:
I tried to do this at first, but I wasn't sure how to do the splatting in the codegen.

Reply:
We can update this later when it's needed. Let's just return the property for now.